lib/routes/kurogames/wutheringwaves/news.ts

```diff
@@ -0,0 +1,58 @@
+import { DataItem, Route } from '@/types';
+import cache from '@/utils/cache';
+import { parseDate } from '@/utils/parse-date';
+import timezone from '@/utils/timezone';
+import * as cheerio from 'cheerio';
+import { ofetch } from 'ofetch';
+
+interface NewsItem {
+    articleContent: string;
+    articleDesc: string;
+    articleId: number;
+    articleTitle: string;
+    articleType: number;
+    createTime: string;
+    sortingMark: number;
+    startTime: string;
+    suggestCover: string;
+    top: number;
+}
+
+export const route: Route = {
+    path: '/wutheringwaves/news',
+    categories: ['game'],
+    example: '/wutheringwaves/news',
+    name: '鸣潮 — 游戏公告、新闻与活动',
+    radar: [
+        {
+            source: ['mc.kurogames.com/m/main/news', 'mc.kurogames.com/main'],
+        },
+    ],
+    maintainers: ['enpitsulin'],
+    description: '',
+    async handler() {
+        const res = await ofetch<NewsItem[]>('https://media-cdn-mingchao.kurogame.com/akiwebsite/website2.0/json/G152/zh/ArticleMenu.json', { query: { t: Date.now() } });
+        return {
+            title: '《鸣潮》— 游戏公告、新闻和活动',
+            link: 'https://mc.kurogames.com/main#news',
+            item: await Promise.all(
+                res.map(async (i) => {
+                    const contentUrl = `https://media-cdn-mingchao.kurogame.com/akiwebsite/website2.0/json/G152/zh/article/${i.articleId}.json`;
+                    const description = (await cache.tryGet(contentUrl, async () => {
+                        const data = await ofetch<NewsItem>(contentUrl, { query: { t: Date.now() } });
+                        const $ = cheerio.load(data.articleContent);
+                        return $('body').html() ?? '';
+                    })) as string;
+
+                    return {
+                        title: i.articleTitle,
+                        pubDate: timezone(parseDate(i.createTime), +8),
+                        description,
+                        link: `https://mc.kurogames.com/main/news/detail/${i.articleId}`,
+                    } as DataItem;
```

Comment:
Cache the whole returned object instead of description only.

Reply:
only fetch the HTML content of this will call multiple times, do we also need to cache other properties that from `https://media-cdn-mingchao.kurogame.com/akiwebsite/website2.0/json/G152/zh/ArticleMenu.json`?

Reply:
By returned object, I mean
https://github.com/DIYgod/RSSHub/blob/673a98a7ecd203fdbd6a5e12cd88695c6301cce5/lib/routes/kurogames/wutheringwaves/news.ts#L47-L52

Reply:
fixed